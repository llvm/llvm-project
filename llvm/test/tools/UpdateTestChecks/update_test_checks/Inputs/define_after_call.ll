; RUN: opt < %s -S | FileCheck %s

define i32 @a() {
  %val = call i32 @b(i32 0)
  ret i32 %val
}

define i32 @b(i32 %arg) {
  ret i32 %arg
}
