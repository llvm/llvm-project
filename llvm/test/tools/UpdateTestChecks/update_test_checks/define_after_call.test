# Verify that calls and defines of the same function can be mixed
# RUN: cp -f %S/Inputs/define_after_call.ll %t.ll && %update_test_checks %t.ll
# RUN: diff -u %t.ll %S/Inputs/define_after_call.ll.expected
# RUN: opt < %t.ll -S | FileCheck %t.ll
