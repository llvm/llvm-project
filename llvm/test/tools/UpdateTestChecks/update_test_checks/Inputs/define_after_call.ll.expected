; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
; RUN: opt < %s -S | FileCheck %s

define i32 @a() {
; CHECK-LABEL: define {{[^@]+}}@a(
; CHECK-NEXT:    [[VAL:%.*]] = call i32 @b(i32 0)
; CHECK-NEXT:    ret i32 [[VAL]]
;
  %val = call i32 @b(i32 0)
  ret i32 %val
}

define i32 @b(i32 %arg) {
; CHECK-LABEL: define {{[^@]+}}@b(
; CHECK-NEXT:    ret i32 [[ARG:%.*]]
;
  ret i32 %arg
}
